salt/modules/vsphere.py

```diff
@@ -3601,6 +3601,991 @@ def vsan_enable(host, username, password, protocol=None, port=None, host_names=N
+    '''
+    log.trace('Building the dict of the DVS \'{0}\' link discovery '
+              'protocol'.format(dvs_name))
+    return {'operation': dvs_link_disc_protocol.operation,
+            'protocol': dvs_link_disc_protocol.protocol}
+
+
+def _get_dvs_product_info(dvs_name, dvs_product_info):
+    '''
+    Returns the dict representation of the DVS product_info
+
+    dvs_name
+        The name of the DVS
+
+    dvs_product_info
+        The DVS product info
+    '''
+    log.trace('Building the dict of the DVS \'{0}\' product '
+              'info'.format(dvs_name))
+    return {'name': dvs_product_info.name,
+            'vendor': dvs_product_info.vendor,
+            'version': dvs_product_info.version}
+
+
+def _get_dvs_capability(dvs_name, dvs_capability):
+    '''
+    Returns the dict representation of the DVS product_info
+
+    dvs_name
+        The name of the DVS
+
+    dvs_capability
+        The DVS capability
+    '''
+    log.trace('Building the dict of the DVS \'{0}\' capability'
+              ''.format(dvs_name))
+    return {'operation_supported': dvs_capability.dvsOperationSupported,
+            'portgroup_operation_supported':
+            dvs_capability.dvPortGroupOperationSupported,
+            'port_operation_supported': dvs_capability.dvPortOperationSupported}
+
+
+def _get_dvs_infrastructure_traffic_resources(dvs_name,
+                                              dvs_infra_traffic_ress):
+    '''
+    Returns a list of dict representations of the DVS infrastructure traffic
+    resource
+
+    dvs_name
+        The name of the DVS
+
+    dvs_infra_traffic_ress
+        The DVS infrastructure traffic resources
+    '''
+    log.trace('Building the dicts of the DVS \'{0}\' infrastructure traffic '
+              'resources'.format(dvs_name))
+    res_dicts = []
+    for res in dvs_infra_traffic_ress:
+        res_dict = {'key': res.key,
+                    'limit': res.allocationInfo.limit,
+                    'reservation': res.allocationInfo.reservation}
+        if res.allocationInfo.shares:
+            res_dict.update({'num_shares': res.allocationInfo.shares.shares,
+                             'share_level': res.allocationInfo.shares.level})
+        res_dicts.append(res_dict)
+    return res_dicts
+
+
+@depends(HAS_PYVMOMI)
+@supports_proxies('esxdatacenter', 'esxcluster')
+@gets_service_instance_via_proxy
+def list_dvss(datacenter=None, dvs_names=None, service_instance=None):
+    '''
+    Returns a list of distributed virtual switches (DVSs).
+    The list can be filtered by the datacenter or DVS names.
+
+    datacenter
+        The datacenter to look for DVSs in.
+        Default value is None.
+
+    dvs_names
+        List of DVS names to look for. If None, all DVSs are returned.
+        Default value is None.
+
+    .. code-block:: bash
+
+        salt '*' vsphere.list_dvss
+
+        salt '*' vsphere.list_dvss dvs_names=[dvs1,dvs2]
+    '''
+    ret_dict = []
```

Comment:
`ret_dict` is a list? This is a bit confusing. :]

Reply:
Old incarnation of the code. Will fix